Source/SwiftLintFramework/Rules/Style/LetVarWhitespaceRule.swift

```diff
@@ -214,12 +219,22 @@ public struct LetVarWhitespaceRule: ConfigurationProviderRule, OptInRule, Automa
     // Collects all the line numbers containing attributes but not declarations
     // other than let/var
     private func attributeLineNumbers(file: SwiftLintFile) -> Set<Int> {
-        return Set(file.syntaxMap.tokens.compactMap({ token in
-            if token.kind == .attributeBuiltin {
-                return file.line(byteOffset: token.offset)
-            }
-            return nil
-        }))
+        let lineNumbers = file.syntaxMap.tokens
+            .filter { isAttribute(token: $0, in: file) }
+            .map(\.offset)
+            .compactMap(file.line)
+        return Set(lineNumbers)
+    }
+
+    private func isAttribute(token: SwiftLintSyntaxToken, in file: SwiftLintFile) -> Bool {
+        let kind = token.kind
+        if kind == .attributeBuiltin {
+            return true
+        }
+        if kind == .typeidentifier, let symbol = file.stringView.substringStartingLinesWithByteRange(token.range) {
```

Comment:
I had guessed that SourceKit was assigning the kind `.attributeId` according to [its documentation](https://github.com/apple/swift/blob/7123d2614b5f222d03b3762cb110d27a9dd98e24/include/swift/IDE/SyntaxModel.h#L54). This does not seems to be the case, however. Instead, this unpleasant check for the `@` prefix must be applied to recognize an attribute.